### DISCUSSION TITLE
Setting MapID?

### DISCUSSION THREAD

**Discussion by itsmeyaw** [JustFly1984/react-google-maps-api#3058] [Q&A] [answered]
Hello, does anybody know where we can set the MapId to be used? I want to use a custom style I already prepared in my Google Cloud Console. 

Those are things I have done:

- Associated my map with the style I want to use
- Setting the parameter `mapId` in `useLoadScript()`
- Setting the `id` parameter in `GoogleMap` component

Somehow I still get the default view. Can someone help me with this?

**Comment by vandervidi** [comment]
Im having the same problem.
I tried also addign the Map ID to the mapIds array  like this. Still not working. 
`<LoadScript
	...
	mapIds={["MAP_ID"]}> ... </LoadScript>`

**Comment by itsmeyaw** [comment, marked as answer]
Heyy, get back to you as there might be anyone still having similar issues. You need to pass the map id inside an `MapOptions` object and pass it to the `options` parameter in your `GoogleMap` component. E.g.,

```js
const options = useMemo<MapOptions>(() => ({
        mapId: mapId
    }), [])

<GoogleMap options={options}/>
```
